dist/performance.js

```diff
@@ -227,60 +239,141 @@ function calcOcclusion(rect){
+            src = splitSrcSet(i.srcset).find(src => src == lcpUrl);
+        }
+
+        return src == lcpUrl;
+    });
+
+    // lcp_elem_stats will match the img, not the source.
+    return summarizeElement(rawLcpElement);
+}
+
+function getLcpResponseObject(lcpUrl) {
+    const responseObject = response_bodies.find(r => {
+        return r.url == lcpUrl;
+    });
+    if (responseObject) {
+        // Don't write the response body to custom metrics.
+        responseObject.response_body = undefined;
+    }
+    return responseObject;
+}
+
 return Promise.all([getLcpElement()]).then(([lcp_elem_stats]) => {
     const lcpUrl = lcp_elem_stats.url;
     const rawDoc = getRawHtmlDocument();
     // Start out with true, only if LCP element is an external resource will we eval & potentially set to false.
     // Let's make sure we're not artificially deflating this metric with LCP elements that aren't external.
     let isLcpStaticallyDiscoverable = true;
     let isLcpPreloaded = null;
-    let responseObject = null;
     let rawLcpElement = null;
-    let gamingMetrics = getGamingMetrics(rawDoc);
+    let lcpPreload = null;
+    let responseObject = null;
+    let gamingMetrics = getGamingMetrics(rawDoc, lcp_elem_stats);
     const isLcpExternalResource = lcpUrl != '';
-    const styles = lcp_elem_stats.styles;
-
-    // Detect LCP Overlay Hack
-    if (styles['pointer-events'] == 'none' &&
-        styles['position'] == 'absolute' &&
-        styles['width'] == '99vw' &&
-        styles['height'] == '99vh') {
-            gamingMetrics['lcpOverlayStrict'] = true;
-    }
-
-    if(lcp_elem_stats.cover90viewport && styles['pointer-events'] == 'none'){
-        gamingMetrics['lcpOverlaySoft'] = true;
-    }
 
     if (isLcpExternalResource) {
         // Check if LCP resource reference is in the raw HTML (as opposed to being injected later by JS)
-        rawLcpElement = Array.from(rawDoc.querySelectorAll('picture source, img')).find(i => {
-            let src = i.src;
-            if (i.nodeName == 'SOURCE') {
-                src = new URL(i.srcset, location.href).href;
-            }
+        rawLcpElement = getRawLcpElement(rawDoc, lcpUrl);
+        lcpPreload = getLcpPreloadInfo(rawDoc, lcpUrl);
+        responseObject = getLcpResponseObject(lcpUrl);
 
-            return src == lcpUrl;
-        });
-        isLcpStaticallyDiscoverable = !!rawLcpElement;
-        isLcpPreloaded = !!Array.from(rawDoc.querySelectorAll('link')).find(link => {
-            return link.rel == 'preload' && link.href == lcpUrl;
-        });
-        responseObject = response_bodies.find(r => {
-            return r.url == lcpUrl;
-        });
-        if (responseObject) {
-            // Don't write the response body to custom metrics.
-            responseObject.response_body = undefined;
-        }
+        isLcpPreloaded = 'header_directives' in lcpPreload || 'tag' in lcpPreload;
+        isLcpStaticallyDiscoverable = !!rawLcpElement || isLcpPreloaded;
```

Comment:
Optional nit: I'm not sure the `!!` is needed here, and I think it makes it a bit less readable.